src/evict/evict_page.c

```diff
@@ -675,12 +675,53 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
-    /* Reconcile the page. */
+    /* Acquire a snapshot if coming through eviction thread route. */
+
+    /*
+     * TODO: We are deliberately not using a snapshot when checkpoint is active. This will ensure
+     * that point-in-time checkpoints have a consistent version of data. Remove this condition once
+     * fuzzy transaction ID based checkpoints work is merged.
+     */
+    if (FLD_ISSET(evict_flags, WT_REC_EVICTION_THREAD) && !WT_IS_HS(S2BT(session)) &&
+      !conn->txn_global.checkpoint_running) {
+        /*
+         * We rely on the fact that the eviction threads are created with read committed isolation
+         * by default. If this fact doesn't hold anymore in future, we have to force isolation
+         * level.
+         */
+        WT_ASSERT(session, session->txn->isolation == WT_ISO_READ_COMMITTED);
+
+        /*
+         * Eviction threads do not need to pin anything in the cache. We have a exclusive lock for
+         * the page being evicted so we are sure that the page will always be there while it is
+         * being processed. Therefore, we use snapshot API that doesn't publish shared IDs to the
+         * outside world.
+         */
+        __wt_txn_bump_snapshot(session);
```

Comment:
Checkpoint status might change before this statement and the above if at 685.

Reply:
@bvpvamsikrishna Good catch! I have pushed 570d9a678 in which I am taking a lock on checkpoints to ensure we never race.